### PR TITLE
Record daily article view count

### PR DESCRIPTION
**Pull request by tomrf1**
## What does this change?
Adds a new local storage item, `gu.history.dailyArticleCount`, which holds counts of articles viewed per day for the past 30 days.

This is required for customised messages in the epic.
There is some additional work to be done before adding this to the epic, as we need to avoid showing it to users with a count below a certain theshold. But we do want to start counting article views now.

Screenshot 2019-05-29 at 16 52 15

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/onward/history.js [hunk 1]
```diff
@@ -462,6 +464,45 @@ const showInMegaNavEnable = (bool: boolean): void => {
...
+        local.set(storageKeyDailyArticleCount, dailyCount);
+    }
+};
+
+const getArticleViewCount = (days: number): number => {
+    const dailyCount = local.get(storageKeyDailyArticleCount) || [];
+    const cutOff = today - days;
+
+    const firstOldDayIndex = dailyCount.findIndex(
+        c => c.day && c.day <= cutOff
```
Comment (jlieb10) [inline static/src/javascripts/projects/common/modules/onward/history.js]: I find the combination of the arrow function and the `less than or equals` on the same line slightly confusing to read. I read the arrow function as 'greater than or equals' at first. Not sure if this is linter-friendly, but would be great if you could move the `c =>` to the line above.

File: static/src/javascripts/bootstraps/enhanced/common.js [hunk 2]
```diff
@@ -151,6 +152,14 @@ const updateHistory = (): void => {
     }
 };
 
+const updateDailyArticleCount = (): void => {
+    const page = config.get('page');
+
+    if (page) {
```
Comment (jlieb10) [inline static/src/javascripts/bootstraps/enhanced/common.js]: Double checking that this function should be called for every `page`. I'm pretty sure this is distinct from a front, but is it definitely an article?

**Comment by jlieb10** [comment]
Great work 👍

File: static/src/javascripts/bootstraps/enhanced/common.js (hunk 2 above)
Comment (tomrf1) [reply static/src/javascripts/bootstraps/enhanced/common.js]: this `page` could be a front, but we check that it's an article inside `incrementDailyArticleCount`

Comment (jlieb10) [reply static/src/javascripts/bootstraps/enhanced/common.js]: 👍

File: static/src/javascripts/projects/common/modules/onward/history.js (hunk 1 above)
Comment (tomrf1) [reply static/src/javascripts/projects/common/modules/onward/history.js]: good point - but unfortunately the linter does not allow it :(

Comment (jlieb10) [reply static/src/javascripts/projects/common/modules/onward/history.js]: 👍

**Review by jlieb10 — APPROVED** [review]

**Review by joelochlann — APPROVED** [review]

**Pull request merged**